libs/service-portal/documents/src/screens/Overview/Overview.tsx

```diff
@@ -57,55 +59,64 @@ type FilterValues = {
   searchQuery: string
 }
 
-const getFilteredDocuments = (
-  documents: Document[],
-  filterValues: FilterValues,
-): Document[] => {
-  const { dateFrom, dateTo, activeCategory, searchQuery } = filterValues
-  let filteredDocuments = documents.filter((document) =>
-    isWithinInterval(new Date(document.date), {
-      start: dateFrom || defaultStartDate,
-      end: dateTo || defaultEndDate,
-    }),
-  )
-
-  if (activeCategory.value) {
-    filteredDocuments = filteredDocuments.filter(
-      (document) => document.senderNatReg === activeCategory.value,
-    )
-  }
-
-  // if (searchQuery) {
-  //   const fuse = new Fuse(filteredDocuments, defaultSearchOptions)
-  //   return fuse.search(searchQuery).map((elem) => {
-  //     // const fuseItem = (elem as unknown) as FuseItem
-  //     // return fuseItem.item
-  //     return elem.item
-  //   })
-  // }
-  if (searchQuery) {
-    return filteredDocuments.filter((x) =>
-      x.subject.toLowerCase().includes(searchQuery.toLowerCase()),
-    )
-  }
-
-  return filteredDocuments
-}
-
 export const ServicePortalDocuments: ServicePortalModuleComponent = ({
   userInfo,
 }) => {
   useNamespaces('sp.documents')
   const { formatMessage, lang } = useLocale()
   const [page, setPage] = useState(1)
   const { scrollToRef } = useScrollToRefOnUpdate([page])
+  const { pathname } = useLocation()
 
   const [filterValue, setFilterValue] = useState<FilterValues>(
     defaultFilterValues,
   )
   const { data, loading, error } = useListDocuments(userInfo.profile.nationalId)
 
   const categories = [defaultCategory, ...data.categories]
+
+  const getFilteredDocuments = (
+    documents: Document[],
+    filterValues: FilterValues,
+  ): Document[] => {
+    const { dateFrom, dateTo, activeCategory, searchQuery } = filterValues
+    let searchInteractionEventSent = false
+    let filteredDocuments = documents.filter((document) =>
+      isWithinInterval(new Date(document.date), {
+        start: dateFrom || defaultStartDate,
+        end: dateTo || defaultEndDate,
+      }),
+    )
+
+    if (activeCategory.value) {
+      filteredDocuments = filteredDocuments.filter(
+        (document) => document.senderNatReg === activeCategory.value,
+      )
+    }
+
+    // if (searchQuery) {
+    //   const fuse = new Fuse(filteredDocuments, defaultSearchOptions)
+    //   return fuse.search(searchQuery).map((elem) => {
+    //     // const fuseItem = (elem as unknown) as FuseItem
+    //     // return fuseItem.item
+    //     return elem.item
+    //   })
+    // }
+    if (searchQuery) {
+      if (!searchInteractionEventSent) {
+        if (pathname) {
+          documentsSearchDocumentsInitialized(pathname)
```

Comment:
Wouldn't this make more sense within the search event listener? `handleSearchChange`

You can also chain booleans instead of nesting if statements:
`if(searchQuery && !searchInteractionEventSent && pathname)`